### PR TITLE
No more shitting meatballs upon consuming protein and flour.

### PR DESCRIPTION
**Pull request by LudwigVonChesterfield**
I have taken a very simple approach to solve #2575, is just outright ban these recipes inside the mob.

Perhaps a better solution would be to force a mob to ingest whatever has been created as a result of such recipe. That I leave for a judgement of somebody else.

### CODE REVIEW

**Comment by GrayRachnid** [comment]
does this stop mixing inaprov and dylo in the body?

**Comment by LudwigVonChesterfield** [comment]
No, should it?

**Comment by GrayRachnid** [comment]
Nop

**Comment by NanakoAC** [comment]
hmmm, i'm not 100% convinced that this is the ideal solution, but it will do. and the feature does seem potentially useful

File: code/modules/reagents/recipes.dm [hunk 1]
```diff
@@ -52,21 +52,30 @@
...
 	if(!holder.has_all_reagents(catalysts))
-		return 0
+		return FALSE
 
 	//check that none of the inhibitors are present in the required amount
 	if(holder.has_any_reagent(inhibitors))
-		return 0
+		return FALSE
+
+	if(require_containers.len && !is_type_in_list(holder.my_atom, require_containers))
```
Comment (NanakoAC) [inline code/modules/reagents/recipes.dm]: does this run istype on all the things in this list?

Comment (LudwigVonChesterfield) [reply code/modules/reagents/recipes.dm]: Yes, exactly.

**Pull request merged**